src/btree/bt_random.c

```diff
@@ -447,10 +483,17 @@ __wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags, WT_
     }
 
     /*
-     * There is no point starting with the root page: the walk will exit immediately. In that case
-     * we aren't holding a hazard pointer so there is nothing to release.
+     * There is no point starting with the root page: continue attempting the process until we
+     * encounter a non-root page.
      */
-    if (!eviction || !__wt_ref_is_root(current))
+    if (eviction && __wt_ref_is_root(current)) {
+        if (--retry > 0)
+            goto restart;
+        else if (!search_inmem_page) {
+            search_inmem_page = true;
```

Comment:
Is search_inmem_page controlled by a flag? We can directly use the inmem algorithm instead of trying 100 times first.

Reply:
search_inmem_page logic only begins if we fail to find a page in 100 tries, I think it is best to start with an inexpensive random search and if we fail then move to a more expensive one, in this case, search_inmem_page.

Reply:
But I think we should not modify the search_inmem_page flag. It should set based on the input flag. If we are not walking inmemory pages and we don't find a page, we should give up immediately.

Reply:
I mean:

else if (F_ISSET(flags, WT_WALK_INMEMORY)) {
          search_inmem_page = true;
}

Reply:
Yes, I will be adding a config to this in coming commits. So that it could be controlled by the user, defaulting to true.

Reply:
I don't think you need a config. The flag should be set by the caller of the function.